enclave-runtime/src/lib.rs

```diff
@@ -412,6 +423,29 @@ fn validate_events(
 		events_proofs.len(),
 		blocks_merkle_roots.len()
 	);
+
+	if events_proofs.len() != blocks_merkle_roots.len() {
+		return Err(Error::ParentChainSync)
+	}
+
+	let events_key = itp_storage::storage_value_key("System", "Events");
+
+	let validated_events: Result<Vec<Vec<u8>>> = events_proofs
+		.iter()
+		.zip(blocks_merkle_roots.iter())
+		.map(|(proof, root)| {
+			StorageProofChecker::<BlakeTwo256>::check_proof(
+				*root,
+				events_key.as_slice(),
+				proof.clone(),
+			)
+			.map_err(|_| Error::ParentChainValidation(itp_storage::Error::WrongValue))
+			.and_then(|opt| opt.ok_or(Error::ParentChainValidation(itp_storage::Error::WrongValue)))
+		})
+		.collect();
```

Comment:
This can probably be a bit simplified.

Reply:
Clippy does not seem to like it either:
```bash
error: use of `ok_or` followed by a function call
   --> src/lib.rs:443:24
    |
443 |             .and_then(|opt| opt.ok_or(Error::ParentChainValidation(itp_storage::Error::WrongValue)))
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `ok_or_else(|| Error::ParentChainValidation(itp_storage::Error::WrongValue))`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
    = note: `-D clippy::or-fun-call` implied by `-D warnings`
```

Reply:
This just because your value in `ok_or` is not a const value, so it prefers the closure variant with `ok_or_else`, which only gets evaluated when you actually go down the error path.